rbtcwrapperproxy/RBTCWrapperProxy.sol

```diff
@@ -227,4 +251,98 @@ contract RBTCWrapperProxy{
+        bool successOfApprove;
+        
+        for (uint256 i = 0; i < lengthOfTokens; i++) {
+            IERC20Token reserveToken = _reserveTokens[i];
+            uint256 reserveAmount = _reserveAmounts[i];
+            successOfTransfer = IERC20Token(reserveToken).transferFrom(msg.sender, address(this), reserveAmount);
+            require(successOfTransfer, "Failed to transfer reserve token from user");
+            successOfApprove = IERC20Token(reserveToken).approve(_liquidityPoolConverterAddress, reserveAmount);
+            require(successOfApprove, "Failed to approve converter to transfer reserve token");
+        }
+
+        (bool successOfDeposit, ) = wrbtcTokenAddress.call.value(_reserveAmounts[lengthOfTokens])(abi.encodeWithSignature("deposit()"));
+        require(successOfDeposit, "Failed to deposit RBTC to get WRBTC");
+        _reserveTokens[lengthOfTokens] = IERC20Token(wrbtcTokenAddress);
+        successOfApprove = IWrbtcERC20(wrbtcTokenAddress).approve(_liquidityPoolConverterAddress, _reserveAmounts[lengthOfTokens]);
+        require(successOfApprove, "Failed to approve converter to transfer WRBTC");
+
+        ILiquidityPoolV1Converter _liquidityPoolConverter = ILiquidityPoolV1Converter(_liquidityPoolConverterAddress);
+        ISmartToken _poolToken = ISmartToken(address(_liquidityPoolConverter.token()));
+        uint256 poolTokenAmountBefore = _poolToken.balanceOf(address(this));
+        _liquidityPoolConverter.addLiquidity(_reserveTokens, _reserveAmounts, _minReturn);
+        uint256 poolTokenAmountAfter = _poolToken.balanceOf(address(this));
+        uint256 poolTokenAmount = poolTokenAmountAfter.sub(poolTokenAmountBefore);
+        
+        successOfTransfer = _poolToken.transfer(msg.sender, poolTokenAmount);
+        require(successOfTransfer, "Failed to transfer pool token to user");
+
+        emit LiquidityAddedToV1(msg.sender, _reserveTokens, _reserveAmounts, poolTokenAmount);
+
+        return poolTokenAmount;
+    }
+
+    
+    function removeLiquidityFromV1(
+        address _liquidityPoolConverterAddress,
+        uint256 _amount, 
+        IERC20Token[] memory _reserveTokens, 
+        uint256[] memory _reserveMinReturnAmounts
+    )   
+        public 
+        checkAddress(_liquidityPoolConverterAddress)
+    {
+        require(_amount > 0, "The amount should larger than zero");
+
+        ILiquidityPoolV1Converter _liquidityPoolConverter = ILiquidityPoolV1Converter(_liquidityPoolConverterAddress);
+        ISmartToken _poolToken = ISmartToken(address(_liquidityPoolConverter.token()));
+
+        bool successOfTransferFrom = _poolToken.transferFrom(msg.sender, address(this), _amount);
+        require(successOfTransferFrom);
+
+        IERC20Token reserveToken;
+        uint256[] memory reserveAmountBefore;
+        for (uint256 i = 0; i < _reserveTokens.length; i++) {
+            reserveToken = _reserveTokens[i];
+            reserveAmountBefore[i] = reserveToken.balanceOf(address(this));
```

Comment:
it will fail during execution due to `reserveAmountBefore` being uninitialized (empty array).
consider initializing it first: `uint256[] memory reserveAmountBefore = uint256[](_reserveTokens.length);`